docs/data/material/components/text-fields/InputSuffixShrink.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import { outlinedInputClasses } from '@mui/material/OutlinedInput';
+import { filledInputClasses } from '@mui/material/FilledInput';
+import { inputBaseClasses } from '@mui/material/InputBase';
+import TextField from '@mui/material/TextField';
+import InputAdornment from '@mui/material/InputAdornment';
+
+export default function InputSuffixShrink() {
+  return (
+    <Box
+      component="form"
+      sx={{ '& > :not(style)': { m: 1, width: '25ch' } }}
+      noValidate
+      autoComplete="off"
+    >
+      <TextField
+        id="outlined-suffix-shrink"
+        label="Outlined"
+        variant="outlined"
+        slotProps={{
+          input: {
+            endAdornment: (
+              <InputAdornment
+                position="end"
+                sx={{
+                  alignSelf: 'flex-end',
+                  opacity: 0,
+                  marginBottom: '5px',
+                  pointerEvents: 'none',
+                  [`.${outlinedInputClasses.root} &`]: {
+                    marginBottom: '16px',
+                  },
+                  [`[data-shrink=true] ~ .${inputBaseClasses.root} > &`]: {
+                    opacity: 1,
+                  },
+                }}
+              >
+                lbs
+              </InputAdornment>
+            ),
+          },
+        }}
+      />
+      <TextField
+        id="filled-suffix-shrink"
+        label="Filled"
+        variant="filled"
+        slotProps={{
+          input: {
+            endAdornment: (
+              <InputAdornment
+                position="end"
+                sx={{
+                  alignSelf: 'flex-end',
+                  opacity: 0,
+                  marginBottom: '5px',
+                  pointerEvents: 'none',
+                  [`.${filledInputClasses.root} &`]: {
+                    marginBottom: '8px',
```

Comment:
The adornment is misaligned by 1 px

```suggestion
                    marginBottom: '7px',
```

Reply:
This is `7px`:

<img width="1703" alt="image" src="https://github.com/user-attachments/assets/d8dcf0f6-6591-4877-bb9d-94e2d1742c79" />


Probably `7.5px` is the best 😅:

<img width="1701" alt="image" src="https://github.com/user-attachments/assets/edc23b04-4867-498a-84e4-1a26ec80f40b" />